app/models/financialDetails/ChargeItem.scala

```diff
@@ -160,14 +165,30 @@ object ChargeItem {
 
   implicit val format: Format[ChargeItem] = Json.format[ChargeItem]
 
-  def filterAllowedCharges(isChargeTypeEnabled:Boolean, chargeType: ChargeType*)
+  def filterAllowedCharges(isChargeTypeEnabled: Boolean, chargeType: ChargeType*)
                           (chargeItem: TransactionItem): Boolean = {
     (isChargeTypeEnabled, chargeItem.transactionType) match {
       case (false, transactionType) if chargeType.toList.contains(transactionType) => false
       case _ => true
     }
   }
 
+  def overdueOrAccruingInterestChargeList(whatYouOweChargesList: WhatYouOweChargesList)(implicit dateServiceInterface: DateServiceInterface): List[ChargeItem] = whatYouOweChargesList.chargesList.filter(x => x.isOverdue() || x.hasAccruingInterest)
```

Comment:
this method is a good candidate to be moved somewhere else as its not using any fields data directly from the ChargeItem instance you are going to call this method from;
maybe worth to create a singleton for WhatYouOweChargesList ?

Reply:
Moved to WhatYouOweChargesList